src/create-glamorous.js

```diff
@@ -42,56 +42,19 @@ function createGlamorous(splitProps) {
        * This is a component which will render the comp (closure)
        * with the glamorous styles (closure). Forwards any valid
        * props to the underlying component.
-       * @param {Object} theme the theme object
-       * @return {ReactElement} React.createElement
        */
-      class GlamorousComponent extends Component {
-        state = {theme: null}
-        setTheme = theme => this.setState({theme})
-
-        componentWillMount() {
-          const {theme} = this.props
-          if (this.context[CHANNEL]) {
-            // if a theme is provided via props,
-            // it takes precedence over context
-            this.setTheme(theme ? theme : this.context[CHANNEL].getState())
-          } else {
-            this.setTheme(theme || {})
-          }
-        }
-
-        componentWillReceiveProps(nextProps) {
-          if (this.props.theme !== nextProps.theme) {
-            this.setTheme(nextProps.theme)
-          }
-        }
-
-        componentDidMount() {
-          if (this.context[CHANNEL] && !this.props.theme) {
-            // subscribe to future theme changes
-            this.unsubscribe = this.context[CHANNEL].subscribe(this.setTheme)
-          }
-        }
-
-        componentWillUnmount() {
-          // cleanup subscription
-          this.unsubscribe && this.unsubscribe()
-        }
-
-        render() {
-          // in this function, we're willing to sacrafice a little on
-          // readability to get better performance because it actually
-          // matters.
-          const props = this.props
+      const GlamorousComponent = withTheme(
+        (props, context) => {
+          /* eslint no-use-before-define: 0 */
           const {toForward, cssOverrides, cssProp} = splitProps(
             props,
             GlamorousComponent,
           )
 
           // freeze the theme object in dev mode
           const theme = process.env.NODE_ENV === 'production' ?
-            this.state.theme :
```

Comment:
We're now a function component so `this` doesn't work and we don't have `state`. `theme` comes from props.